### DISCUSSION TITLE
Rotary dial doesn't work when the LED is plugged in

### DISCUSSION THREAD

**Discussion by aconbere** [srozum/film_camera_tester#42] [Q&A]
[Edit - Initial I thought I had disconnected the LED but it wasn't fully loose - edited the body to show that]

When I have the light source connector plugged into the control board AND the LED connected the rotary dial doesn't work. The switch works, and all other facilities seem to work. The LED and fan turn on when a test is running, and turn off when it stops. When the LED is unplugged the rotary dial works beautifully.

~~I've tried disconnecting the fan, and disconnecting the LED, and disconnecting the 12v source, but none of these change the behavior of the issue.~~

Based on my last question (#40) I double checked that all the wires have connectivity to the correct pins on the arduino (they do).

Thoughts on next places to look to debug?

My assumption is that the arduino has a software grey code decoder in it that's watching the bounce from the rotary dial. Is this potentially another noise floor issue?

**Comment by aconbere** [comment]
I lied! I hadn't fully disconnected the LED. When the LED is disconnected the rotary dial works. Will fix the post.

**Comment by srozum** [comment]
I am sorry, but this explanation makes no sense to me. Make sure that LightUnit connected to socket BEFORE power on, you cannot plug or unplug LightUnit during operation.

**Reply by aconbere** [reply]
@srozum sorry for any confusion.

When i'm saying that I've connected or disconnected various parts I mean that i've powered off everything and unplugged its connector.

So the reason I'm mucking about with connectors is just trying to isolate what if any specific component might be causing the issue.

At this point I can connect the light source, with everything except the LED plugged into it, and the rotary dial works. If i power everything off, attach the led, power things back on, the rotary dial will not function.

**Reply by srozum** [reply]
Still, not enough information. There is no connection between LightUnit (is that what you mean by LED?) and rotary switch. Again, check your wiring and soldering job.

**Reply by aconbere** [reply]
Hmm when I say LED I mean the component of the Light Unit. On mine I've attached the LED to the light unit board with a JST connector and I can just unplug it.

I agree that there is or should be no relationship. Was hoping to you might have a snappy idea like the last issue :)

I'll keep digging
